File: javascript/9_Palindrome_Number.js
// 1. Reverse String
var isPalindrome = function(x) {
    // 1) x is negative  2)x is at least two digits，the single digit of x is 0
    if (x < 0 || (x % 10 === 0 && x !== 0)) {
        return false;
    }
    let s = String(x);
    rs = '';
    for (let i = s.length-1; i>=0; i--) {
        rs += s[i];
    }
    return s === rs;
};

output1 = isPalindrome(121);
output2 = isPalindrome(-121);
output3 = isPalindrome(10);
console.log(output1, output2, output3);

// 2.
var isPalindrome2 = function(x) {
    
}

File: python/9_Palindrome_Number.py
# 1
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        reverse_num = 0
        while x > reverse_num:
            reverse_num = 10 * reverse_num + (x % 10)
            x //= 10

        return x == reverse_num or x == reverse_num//10

output1 = Solution().isPalindrome(121)
output2 = Solution().isPalindrome(-121)
output3 = Solution().isPalindrome(10)

print(output1, output2, output3)

# 2
class Solution2:
    def isPalindrome(self, x: int) -> bool:
        s = str(x)
        return s == s[::-1]  # str[<start>:<stop>:<step>]
